full/src/test/java/apoc/redis/RedisTest.java

```diff
@@ -0,0 +1,375 @@
+        Object keyEval = getByCodec("testEval");
+        TestUtil.testCall(db, "CALL apoc.redis.setGet($uri, $keyEval, $valueEval, $config)",
+                map("uri", URI, "keyEval", keyEval, "valueEval", getByCodec("valueEval"), "config", map("codec", codec)), 
+                r -> assertNull(r.get("value")));
+        
+        TestUtil.testCall(db, "CALL apoc.redis.eval($uri, 'return redis.call(\"get\", KEYS[1])', 'VALUE', [$keyEval], [$keyName], $config)", 
+                map("uri", URI, "keyEval", keyEval, "keyName", getByCodec("key:name"), "config", map("codec", codec)),
+                r -> assertEquals("valueEval", fromCodec(r.get("value"))));
+    }
+
+    @Test
+    public void testKeysCommand() {
+        Map<String, Object> config = map("codec", this.codec);
+        Object from = getByCodec("from");
+        Object to = getByCodec("to");
+        TestUtil.testCall(db, "CALL apoc.redis.setGet($uri, $from, $one, $config)",
+                map("uri", URI, "from", from, "one", getByCodec("one"), "config", config),
+                r -> assertNull(r.get("value")));
+
+        TestUtil.testCall(db, "CALL apoc.redis.copy($uri, $from, $to, $config)",
+                map("uri", URI, "from", from, "to", to, "config", config),
+                r -> assertEquals(true, r.get("value")));
+        
+        TestUtil.testCall(db, "CALL apoc.redis.get($uri, $to, $config)",
+                map("uri", URI, "to", to, "config", config),
+                r -> assertEquals("one", fromCodec(r.get("value"))));
+        
+        TestUtil.testCall(db, "CALL apoc.redis.exists($uri, [$to], $config)",
+                map("uri", URI, "to", to, "config", config),
+                r -> assertEquals(1L, r.get("value")));
+        
+        TestUtil.testCall(db, "CALL apoc.redis.pexpire($uri, $to, 9999, false, $config)",
+                map("uri", URI, "to", to, "config", config),
+                r -> assertEquals(true, r.get("value")));
+        
+        TestUtil.testCall(db, "CALL apoc.redis.pttl($uri, $to, $config)",
+                map("uri", URI, "to", to, "config", config),
+                r -> {
+                    final long value = (long) r.get("value");
+                    assertTrue( value <= 9999L && value > 0L);
+                });
+
+        TestUtil.testCall(db, "CALL apoc.redis.persist($uri, $to, $config)",
+                map("uri", URI, "to", to, "config", config),
+                r -> assertEquals(true, r.get("value")));
+        
+        TestUtil.testCall(db, "CALL apoc.redis.pttl($uri, $to, $config)",
+                map("uri", URI, "to", to, "config", config),
+                r -> assertEquals(-1L, (long) r.get("value")));
+    }
+
+    @Test
+    public void testServersCommand() {
+        TestUtil.testCall(db, "CALL apoc.redis.info($uri, $config)",
+                map("uri", URI, "config", map("codec", codec)),
+                r -> assertTrue(((String) r.get("value")).contains("redis_version:6.2.3")));
```

Comment:
maybe we can be a bit less specific about the version to make upgrades easier?

Reply:
Now I extracted the string the `6.2.3` string here and in `GenericContainer` param,
in a `REDIS_VERSION` constant to avoid discrepancies.